<launch>
  <arg name="rviz" default="true" doc="Should rviz be spawned?" />
  <arg name="controller" default="cartesian_impedance_example_controller" />

  <arg name="object_x" default="0.0" />
  <arg name="object_y" default="0.4" />
  <arg name="object_z" default="0.02" />

  <arg name="scene_xml" default="$(find dual_arm_control)/worlds/dual_panda_with_object.xml.xacro" />
  <arg name="use_xacro" default="true" />

  <arg name="use_suction_gripper_0" default="false" />
  <arg name="use_suction_gripper_1" default="true" />

  <param name="/is_simulation" value="true" />

  <include file="$(find franka_mujoco)/launch/dual_panda.launch">
    <arg name="rviz" value="$(arg rviz)" />
    <arg name="use_gripper" value="true" />
    <arg name="controller" value="$(arg controller)" />
    <arg name="unpause" value="true" />
    <arg name="scene_xml" value="$(arg scene_xml)" />
    <arg name="interactive_marker" value="false" />
    <arg name="use_sim_time" value="true" />
    <arg name="use_xacro" value="true" />
    <arg name="use_suction_gripper_0" value="$(arg use_suction_gripper_0)" />
    <arg name="use_suction_gripper_1" value="$(arg use_suction_gripper_1)" />
  </include>

  <node name="handover_sequence_controller"
        pkg="dual_arm_control"
        type="handover_sequence.py"
        output="screen"
        launch-prefix="bash -c 'sleep 5; $0 $@' ">
    <param name="object_pos_x" value="$(arg object_x)" />
    <param name="object_pos_y" value="$(arg object_y)" />
    <param name="use_suction_gripper_0" value="$(arg use_suction_gripper_0)" />
    <param name="use_suction_gripper_1" value="$(arg use_suction_gripper_1)" />
  </node>

</launch>
